libcontainer/cgroups/fs2/memory.go

```diff
@@ -78,7 +78,7 @@ func setMemory(dirPath string, r *configs.Resources) error {
 	return nil
 }
 
-func statMemory(dirPath string, stats *cgroups.Stats) error {
+func statMemory(rootCgroupPath, dirPath string, stats *cgroups.Stats) error {
```

Comment:
I don't love this, but it made unit testing simpler and keeps all exposed APIs unchanged.

Welcome better testing suggestions, especially for some way to diff v1/v2? I'd need some bigger changes to the manager for that.

Reply:
We could make `UnifiedMountpoint` a variable that you can override in tests. We do this for some other stuff in the cgroup tests. The only issue is that it is publicly exposed and we probably don't want to expose allowing this to be changed by libcontainer users...

Reply:
would you prefer that?

I think this signature is awkward, but could probably be better resolved by more deliberate refactoring separate from this issue, while avoiding an exported/mutable var like you mentioned.

I was digging into the FS calls to see if I could drop in a mock, but quickly realized that wouldn't be so simple 😄

Reply:
It seems the only point of adding this extra argument is to make the function call `statsFromMeminfo`. If that's the case, we can test `statsFromMeminfo` directly in the unit tests.

Reply:
ah good call. I left a test for pod cgroup as an "anti-test" on the root cgroup, and added a simple test for v2 root, although I hoped to avoid mocking swap calculations.